pandas/_libs/tslibs/period.pyx

```diff
@@ -2728,6 +2750,11 @@ class Period(_Period):
             freq = cls._maybe_convert_freq(freq)
         nanosecond = 0
 
+        if freq.__class__ in PERIOD_DEPR_OFFSETS:
+            raise ValueError(
+                f"{(type(freq).__name__)} is not supported as period frequency"
+            )
```

Comment:
instead of hard-coding the offsets and checking this, could we just do
```python
if not hasattr(freq, '_period_dtype_code'):
    raise ValueError(...)
```
?

Reply:
Thank you, I corrected the condition and the error message. We have `_period_dtype_code` in class `CustomBusinessDay` (see [link](https://github.com/pandas-dev/pandas/pull/52567)) that is why I excluded CustomBusinessDay.

I thought, isn't better to use `TypeError` instead of `ValueError`?
Should we add a note to v2.2.0? Something like `Bug in incorrectly allowing construction of :class:'Period' ...`

Reply:
thanks - yeah `TypeError` is probably better

and yes, a 2.2.0 whatsnew note would be good

maybe we can just do
```
try:
    period_dtype_code = freq._period_dtype_code
except (AttributeError, TypeError):
    # AttributeError: _period_dtype_code might not exist
    # TypeError: _period_dtype_code might intentionally raise
    raise TypeError(f"{(type(freq).__name__)} is not supported as period frequency"
```
and then reuse `period_dtype_code` further down where there's `freq._period_dtype_code`?

Reply:
thanks, I did as you suggested. It works very well.